### ISSUE TITLE
JAWS radio bug - Interaction bug and visual focus indicator alignment bug

### ISSUE THREAD

**Issue by jasonday** [FreedomScientific/VFO-standards-support#537]
Alignment issue video:
https://user-images.githubusercontent.com/748011/120837993-55e5e380-c535-11eb-975b-d40bbdb28673.mp4

To reproduce:
* navigate radio options

Selection/interaction issue videos:
https://user-images.githubusercontent.com/748011/120846173-bed25900-c53f-11eb-94ec-23141a0d0086.mp4

To reproduce:
* JAWS on
* Navigate to radio group
* Arrow through radios to last radio and select using SPACE
* TAB away from radio group
* SHIFT + TAB back to radio group
* Arrow up to first radio button in group
  * Note: the browser focus indicator remains on the last radio button, and the JAWS visual focus indicator is now on the first radio button
* hit TAB
  * Browser focus and JAWS focus indicators are now on the second (or next) radio button

Which was captured on this page (as well as identified on other radio implementations) :
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio

Details:
* JAWS 2020
* Chrome 90.0.4430.212

**Comment by stevefaulkner** [comment]
@jasonday appears to be issue with controls in `iframe`s Also appears to be chrome issue as it works as expected in Firefox, I suspect it may be due to incorrect position co-ordinates being conveyed by chrome in acc layer, have seen a similar issue in past with chrome, but it was fixed. Will dig into it more.

**Comment by jasonday** [comment]
@stevefaulkner that makes sense for the first issue. 

For the second (and potentially bigger issue) we are seeing similar behavior in non-iframed controls. 
You can recreate the issue across a wide range of radio examples. 
* https://pattern-library.dequelabs.com/components/radio-buttons
* https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/radio/radio-1/radio-1.html

**Comment by stevefaulkner** [comment]
@thanks will check this, note that For the purposes of issues we always check in latest version of JAWS (currently 2021) if the issue is not present in latest version its considered fixed.

**Comment by aardrian** [comment]
@jasonday Does this video correspond to your experience?

https://user-images.githubusercontent.com/1376607/120858718-30ff6980-c551-11eb-9be9-5af49624f471.mp4

Note in Firefox when I `Shift`+`Tab` while on the radio group, focus moves up the page (as expected, but which is also outside the video viewport). I could only re-create it in Chrome. I used JAWS 2020 with Chrome 91 and Firefox 88.

**Comment by jasonday** [comment]
@aardrian Yes, that looks accurate to what I am seeing in testing as well.

**Comment by dashouse** [comment]
Hi, I appreciate this issue is a couple of years old but I believe we are encountering something similar and I was wondering if you were able to resolve / confirm the expected behaviour.

These are technically radios presented as selector buttons.

NVDA in Edge, Chrome, Firefox: works as expected. SR focus moves with visual focus.

JAWS in Edge, Chrome: tab into radio group, traverse with arrow key but visual focus does not move with Jaws focus.

Apologies for the poor screenshot - my org is incredibly locked down and I had to do this on my phone

![IMG_7780](https://github.com/FreedomScientific/standards-support/assets/23356842/58c25ebf-43b0-4c4b-b6c5-25acbffd3c43)

**Comment by JAWS-test** [comment]
@dashouse 

Your photo looks as if you are not in forms mode. Did you activate the radio button with Enter and then navigate with the arrow keys? JAWS is not automatically in forms mode when focusing on a radio button - but operation with the arrow keys is only possible in forms mode. Your photo looks as if you are just navigating with the virtual cursor. Then the behavior and display (of the focus) would be correct

**Comment by dashouse** [comment]
@JAWS-test yes, the screenshot shows the first item both as checked and visually focused. When checked / unchecked the behaviour appears to be the same.

If this is unexpected I will look deeper into our implementation. It just looked so familiar to the examples on this issue I was wondering if it was something specific that might be causing it.

**Comment by JAWS-test** [comment]
@dashouse When a radio button is activated (with Enter), JAWS switches to form mode and navigation with the arrow keys between the radio buttons should be possible. If this does not work in your example, the source code is incorrect or there is a JAWS bug. This can only be analyzed if you send the source code. You can check the normal JAWS behavior for radio buttons yourself at `<input type=radio>` (e.g. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)

**Comment by dashouse** [comment]
@JAWS-test thanks for the advice. We believe it is something in our implementation that caused the behaviour to change between JAWS versions. Testing on an old version (18) our implementation works as expected but from 19 it does not. This narrows it down enough for us to figure out I'm sure!
